### PR TITLE
Validate Adagrad initial_accumulator_value is non-negative

### PR DESCRIPTION
**Pull request by kaivalya-cyber**
`Adagrad` currently accepts a negative `initial_accumulator_value`. Accumulators only grow over time (they accumulate squared gradients), so a negative starting value is invalid and produces undefined optimizer behavior instead of a clear error.

This adds a `ValueError` at construction, matching TF-Keras 2's `Adagrad`, which has always rejected negative values:

```python
>>> keras.optimizers.Adagrad(initial_accumulator_value=-0.1)
ValueError: Argument `initial_accumulator_value` must be a non-negative float. Received: initial_accumulator_value=-0.1
```

Non-negative values, including `0.0`, remain accepted, and a regression test covers both the rejection and the boundary case.

## Contributor License Agreement

- [X] I have read and followed the [contributor guidelines](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md).
- [X] This pull request is my own work, and I agree to the terms of the [Keras CLA](https://cla.developers.google.com/).

### CODE REVIEW

File: keras/src/optimizers/adagrad.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ def __init__(
             name=name,
             **kwargs,
         )
+        if initial_accumulator_value < 0.0:
```
Comment (mathewOracle) [inline keras/src/optimizers/adagrad.py]: initial_accumulator_value=-0.1 with a gradient
of 1.0 actually recovers fine, because the accumulator climbs to 0.9 before
the sqrt happens. It's only the larger negative values that break it, which
is probably why this hasn't been caught by casual testing before.

Style-wise this matches the existing precedent in adamw.py (raising after
super().__init__(), same "Argument `x` must be ... Received: x={x}" message
shape), so it's consistent with the rest of the optimizers

**Comment by JyotinderSingh** [comment]
Please sign the contributor agreement as expected in the PR template and ensure that the CI is green.
